File: api/RestaurantBusiness/Models/Food.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace RestaurantBusiness.Models
{
    public class Food
    {
        public string Name { get; set; }

        public decimal Cost { get; set; }

        public List<Ingredient> Ingredients { get; set; }    
    }
}

File: angular/restaurant-business/src/models/food.ts
import { Ingredient } from './ingredient';

export class Food {
    constructor(name: string, cost: number, ingredients: Ingredient[]) {
        this.name = name;
        this.cost = cost;
        this.ingredients = ingredients;
    }
    name: string;
    cost: number;
    ingredients: Ingredient[];
}